eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/location/list/LocationSuggestionItemView.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.location.list;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.schedjoules.eventdiscovery.R;
+
+
+/**
+ * View for the location suggestion items in the list.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class LocationSuggestionItemView extends LinearLayout
+{
+
+    private TextView mTitle;
+    private TextView mSubTitle;
+
+
+    public LocationSuggestionItemView(Context context, AttributeSet attrs)
+    {
+        super(context, attrs);
+    }
+
+
+    @Override
+    protected void onFinishInflate()
+    {
+        super.onFinishInflate();
+        mTitle = (TextView) findViewById(R.id.schedjoules_location_suggestion_item_title);
+        mSubTitle = (TextView) findViewById(R.id.schedjoules_location_suggestion_item_subtitle);
+    }
+
+
+    public void setTitle(CharSequence title)
```

Comment:
You are right that it wasn't clear how I would approach this smart or dumb view for the list items, because I felt that the `ListItem` itself usually need to hold the data anyway, item views are also recycled, so it might be better to keep data and logic there. But I like that `SmartView` interfaces as well, also to be coherent with other (not list item) views, this would be a clearer approach.
I've added it and I see that it can lead to more generalization of the `ListItem`s, I will look into that in the next round. Namely to check if `ListItem<V extends View>` can be changed to `ListItem<V extends SmartView>` and also to check how to handle callback if needed.
A `SmartView` can of course execute stuff itself, but I am not sure if it should execute something async and then act on the result, that clearly seems too much, so in some cases we might still need callback from it. It depends also on whether that loading can be done in a new `MicroFragment`, then it's not a problem. For now, I left the callbacks in, which also meant that `PlaceSuggestionItem` is still very specific.

About making `PlaceSuggestionItemView` `TwoLineItemView`, I think the layout will be usually specific to the actual place of usage, so my feeling is that it's probably not worth trying to create something general, it might just cause problem if we used something like that for multiple places and couple them that way, we couldn't change one without changing the other.